### DISCUSSION TITLE
Question about the new Rate limiting

### DISCUSSION THREAD

**Discussion by rbartholomay** [adonisjs/core#3819] [Help] [answered]
Hi,

currently playing with the new Rate-Limiting. In the docu there is following:

> Support for multiple storage backends. Redis, PostgreSQL/MySQL and **Memory**.

Is "Memory" really supported? I only found configs for Redis and MySQL...

Have a great day!

René

**Comment by Barbapapazes** [comment, marked as answer]
Yes, it's could be an error? [Because in code](https://github.com/adonisjs/limiter/blob/25e63197b2047410b02c9f81710b13c53a514da1/src/contracts/index.ts#L50), only Redis and DB is shown!

But, you can use SQLite ([supported in Adonis](https://docs.adonisjs.com/guides/database/introduction#drivers-config))
> SQLite in-memory databases are databases stored entirely in memory, not on disk. [Source 1](https://docs.microsoft.com/en-us/dotnet/standard/data/sqlite/in-memory-databases) [Source 2](https://www.sqlite.org/inmemorydb.html)

**Reply by davidharting** [reply]
I think by "memory", @rbartholomay is referring to an in-memory cache in the Node.js process (basically, a global variable). So I don't think using in-memory sqlite addresses OP's question here.

**Reply by Barbapapazes** [reply]
Ah, maybe! Only @thetutlage know this.

**Reply by thetutlage** [reply]
Nope, there is going to be a memory driver that can mainly be used during testing. I expected to work on it right after the initial release, but got delayed.

**Reply by maximehuylebroeck** [reply]
> Nope, there is going to be a memory driver that can mainly be used during testing. I expected to work on it right after the initial release, but got delayed.

I created a PR for this already :) (https://github.com/adonisjs/limiter/pull/1)
Was a quick proposal as I mainly wanted it as a mock for testing, but it could be edited so we can use it as a normal in memory driver but there will be a caveat for users that deploy with clustering. Because obviously they will all have different memory pools. If we want to support that we need to add the cluster driver as well.

**Reply by thetutlage** [reply]
Unless there is a real need. I will not expose the `memory` driver to be in normal configuration. In prod, one should be using Redis mainly.

**Reply by maximehuylebroeck** [reply]
Could be helpful in the development phase though, anyways I have it ready in a branch if someone wants the in memory driver.
